src/lib/GasTank.ts

```diff
@@ -15,28 +19,89 @@ export class GasTank {
 
     // private fields
     #relayer: BiconomyRelayer; // We can simply swap out biconomy by using a different relayer
+    #authorizer: QuestbookAuthorizer; // We can change the authorizer by simply swapping out the QuestbookAuthorizer
 
     constructor(gasTank: GasTankProps, databaseConfig: DatabaseConfig) {
         this.gasTankName = gasTank.name;
         this.chainId = gasTank.chainId;
+        this.#relayer = new BiconomyRelayer({
+            name: gasTank.name,
+            chainId: gasTank.chainId,
+            provider: new ethers.providers.JsonRpcProvider(gasTank.providerURL),
+            apiKey: gasTank.apiKey,
+            providerURL: gasTank.providerURL
+        });
+        this.#authorizer = new QuestbookAuthorizer(databaseConfig);
+    }
+
+    async buildExecTransaction(
+        populatedTx: string,
+        targetContractAddress: string,
```

Comment:
Create a new type `BuildExecTransactionParams` and encapsulate these params inside it, so that when we will swap biconomy we simply change that type.

Reply:
Check how it's done with `sendGaslessTransaction`